site/src/components/JsonSchemaPreview.tsx

```diff
@@ -346,6 +382,17 @@ function resolveSchema(schema: JSONSchema7, defs: Record<string, JSONSchema7Defi
     throw new Error('unsupported');
   }
 
+  if (schema.anyOf) {
+    return {
+      ...schema,
+      anyOf: schema.anyOf.map(s => resolveSchema(s as JSONSchema7, defs))
+    };
+  }
```

Comment:
`resolveSchema` only acts on a single schema at a time, and is always called at the start of `Schema` so  it doesn't have to be used on `anyOf` since it gets taken care of later